### ISSUE TITLE
batch_norm_elemt_cuda_template does not use its argument epsilon

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#27466]
In the internal function [batch_norm_elemt_cuda_template](
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Normalization.cuh#L672) epsilon is passed to [batch_norm_transform_input_kernel](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cuda/Normalization.cuh#L159), but `train=true` template parameter is used, so in fact epsilon is not used there either.

**Comment by pietern** [comment]
Thanks for reporting.

I'm not sure what the right resolution is here. Tagging for triage review.

**Comment by vadimkantorov** [comment]
I think the resolution can be removing `epsilon` argument from `batch_norm_elemt_cuda_template` and `torch.batch_norm_elemt` (since it's not used)

**Comment by SsnL** [comment]
I feel that this should be labeled as high pri bug instead of BC breaking.

**Comment by gchanan** [comment]
Unfortunately we expose this function in the top-level namespace and it's listed in the JIT documentation.  So we probably want to deprecate it (and we should remove it from the top-level namespace, it's not documented and the name is not good).

**Comment by gchanan** [comment]
@SsnL I just wrote why I think it's BC breaking.

I don't see why it's high priority, though.  It's strange behavior for a function that IMO shouldn't even be exposed.

**Comment by SsnL** [comment]
I see.  I thought this is part of the batch norm call and this would be a
pretty bad bug. Probably I am wrong.

**Comment by vadimkantorov** [comment]
There is no bug, eps is taken care of in the caller, it's just confusing when reading the code.

@gchanan This function is used in SyncBN and maybe for some JIT un-lowering to do fusion. I have a [mini-PR](https://github.com/pytorch/pytorch/pull/27621) to use it for a first step towards inplace batchnorm.

**Comment by gchanan** [comment]
that PR looks fine, it doesn't affect the python API because the out variant gets collapsed there.
